### PR TITLE
[Keyboard] Added support for Aves 65

### PR DESCRIPTION
**Pull request by Hund**
## Description

Support for the board Aves 65.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

I'm not sure what the second last point means?

### CODE REVIEW

File: keyboards/aves65/aves65.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    { k00, k01, k02, k03, k04, k05, k06, k07, k08, k09, k0a, k0b, k0c, k0d, k0e }, \
+    { k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, k1b, k1c, ___, k1e }, \
+    { k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b, k2c, k2d, k2e }, \
+    { k30, k31, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b, k3c, k3d, k3e }, \
+    { k40, k41, k42, ___, ___, ___, k46, ___, ___, ___, k4a, k4b, k4c, k4d, k4e }  \
+}
+
+
+void matrix_init_user(void);
+void matrix_scan_user(void);
```
Comment (drashna) [inline keyboards/aves65/aves65.h:23]: These shouldn't be needed here.

```suggestion
```

File: keyboards/aves65/rules.mk [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+COMMAND_ENABLE = no     # Commands for debug and configuration
+NKRO_ENABLE = no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+RGBLIGHT_ENABLE = yes   # Enable keyboard underlight functionality (+4870)
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality (+1150)
+MIDI_ENABLE = no 		# MIDI controls
+AUDIO_ENABLE = no
+UNICODE_ENABLE = no 		# Unicode
+BLUETOOTH_ENABLE = no # Enable Bluetooth with the Adafruit EZ-Key HID
+
+LAYOUTS = 65_ansi 65_iso
```
Comment (drashna) [inline keyboards/aves65/rules.mk:30]: You don't have a layout macro for `65_ansi` defined.  You should either remove support for it, or add the layout.

```suggestion
LAYOUTS = 65_iso
```

File: keyboards/aves65/rules.mk [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = no	# Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
+CONSOLE_ENABLE = no  	# Console for debug(+400)
+COMMAND_ENABLE = no     # Commands for debug and configuration
+NKRO_ENABLE = no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+RGBLIGHT_ENABLE = yes   # Enable keyboard underlight functionality (+4870)
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality (+1150)
```
Comment (fauxpark) [inline keyboards/aves65/rules.mk:24]:
```suggestion
BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
MOUSEKEY_ENABLE = no	# Mouse keys
EXTRAKEY_ENABLE = yes	# Audio control and System control
CONSOLE_ENABLE = no  	# Console for debug
COMMAND_ENABLE = no     # Commands for debug and configuration
NKRO_ENABLE = no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
RGBLIGHT_ENABLE = yes   # Enable keyboard underlight functionality
BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
```

File: keyboards/aves65/config.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4705
+#define PRODUCT_ID      0x4565
```
Comment (fauxpark) [inline keyboards/aves65/config.h:24]:
```suggestion
#define VENDOR_ID       0xFEED
#define PRODUCT_ID      0x9038
```

```
echo -n "Aves65" | shasum
9038671679429c30ff76ea1db04da9dc55e67dfa  -
```

File: keyboards/aves65/rules.mk [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no  	# Console for debug
+COMMAND_ENABLE = no     # Commands for debug and configuration
+NKRO_ENABLE = no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+RGBLIGHT_ENABLE = yes   # Enable keyboard underlight functionality
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+MIDI_ENABLE = no 		# MIDI controls
+AUDIO_ENABLE = no
+UNICODE_ENABLE = no 		# Unicode
+BLUETOOTH_ENABLE = no # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (zvecr) [inline keyboards/aves65/rules.mk:28]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

File: keyboards/aves65/aves65.h [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include "quantum.h"
+
+// readability
+#define ___ KC_NO
+#define LAYOUT_65_iso( \
```
Comment (zvecr) [inline keyboards/aves65/aves65.h:6]:
```suggestion
#define LAYOUT( \
```

File: keyboards/aves65/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _MA 0
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+[_MA] = LAYOUT_65_iso(
```
Comment (zvecr) [inline keyboards/aves65/keymaps/default/keymap.c:11]:
```suggestion
[_MA] = LAYOUT(
```

File: keyboards/aves65/rules.mk [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+
+LAYOUTS = 65_iso
```
Comment (zvecr) [inline keyboards/aves65/rules.mk:34]:
```suggestion
```

**Comment by noroadsleft** [comment]
> Due to only 2 right mods, this is not community layout compatible.

Actually, looking at [this photo](https://fosstodon.b-cdn.net/media_attachments/files/004/089/184/original/d28e3a5a53104030.jpg), it may be a [`LAYOUT_65_iso_blocker`](https://github.com/qmk/qmk_firmware/tree/52af38da37078ad55095734fe7e4eb2c813b3ca5/layouts/default/65_iso_blocker) board.

**Comment by Hund** [comment]
>Actually, looking at this photo, it may be a LAYOUT_65_iso_blocker board.

It's a hotswap PCB. I'm just the person financing this, I'm not the brain behind this so I'm not familiar with all the lingo. Or what it means for the PCB to be a community layout or whatnot.

I just assumed the firmware I was provided would be enough, apparently not. I would like to help as much as I can, but I don't follow the lingo.

**Comment by noroadsleft** [comment]
> It's a hotswap PCB. I'm just the person financing this, I'm not the brain behind this so I'm not familiar with all the lingo. Or what it means for the PCB to be a community layout or whatnot.

Community Layouts TL;DR: If a user owns multiple keyboards with the same physical layout, [QMK has a feature](https://docs.qmk.fm/#/feature_layouts) where the user can write one `keymap.c` file and use that same keymap on any compatible keyboard.

Is the assembled layout for the board (key sizes and positions) [the same as this?](http://www.keyboard-layout-editor.com/##@@_a:7%3B&=&=&=&=&=&=&=&=&=&=&=&=&=&_w:2%3B&=&=%3B&@_w:1.5%3B&=&=&=&=&=&=&=&=&=&=&=&=&=&_x:0.25&w:1.25&h:2&w2:1.5&h2:1&x2:-0.25%3B&=&=%3B&@_w:1.75%3B&=&=&=&=&=&=&=&=&=&=&=&=&=&_x:1.25%3B&=%3B&@_w:1.25%3B&=&=&=&=&=&=&=&=&=&=&=&=&_w:1.75%3B&=&=&=%3B&@_w:1.25%3B&=&_w:1.25%3B&=&_w:1.25%3B&=&_w:6.25%3B&=&_w:1.25%3B&=&_w:1.25%3B&=&_x:0.5%3B&=&=&=)

**Comment by Hund** [comment]
Here's some changes. I'm not sure how to attach them other than with this link. I hope it's alright.

https://github.com/Hund/qmk_firmware/commit/722a3417d11236384bce0659e142bf4c9af012d5

File: keyboards/aves65/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _MA 0
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+<<<<<<< HEAD
```
Comment (zvecr) [inline keyboards/aves65/keymaps/default/keymap.c:11]: You have a bad merge here.

File: keyboards/aves65/rules.mk [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = qmk-dfu
```
Comment (zvecr) [inline keyboards/aves65/rules.mk:12]: Does this board run qmk-dfu by default?

Comment (Hund) [reply keyboards/aves65/rules.mk:12]: Yes.

File: keyboards/aves65/info.json [hunk 11]
```diff
@@ -0,0 +1,85 @@
+{
+    "keyboard_name": "Aves65", 
+    "url": "", 
+    "maintainer": "Hund", 
+    "width": 16, 
+    "height": 5, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/aves65/info.json:8]:
```suggestion
        "LAYOUT_65_iso_blocker": {
```

File: keyboards/aves65/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+[_MA] = LAYOUT_65_iso_blocker(
+=======
+[_MA] = LAYOUT(
+>>>>>>> 8ab76dec206113306916cc5e4298b3f563d6ffa7
```
Comment (noroadsleft) [inline keyboards/aves65/keymaps/default/keymap.c:15]:
```suggestion
[_MA] = LAYOUT_65_iso_blocker(
```

File: keyboards/aves65/readme.md [hunk 13]
```diff
@@ -0,0 +1,15 @@
+# Aves65
+
+![Aves65](https://i.imgur.com/H7x8an7.jpg)
```
Comment (noroadsleft) [inline keyboards/aves65/readme.md:3]:
```suggestion
![Aves65](https://i.imgur.com/H7x8an7l.jpg)
```

File: keyboards/aves65/readme.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+
+A 65% hotswap keyboard with RGB underglow
```
Comment (noroadsleft) [inline keyboards/aves65/readme.md:5]:
```suggestion
A 65% hotswap keyboard with RGB underglow.
```

File: keyboards/aves65/rules.mk [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (zvecr) [inline keyboards/aves65/rules.mk:32]:
```suggestion
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs

LAYOUTS = 65_iso_blocker
```

**Comment by zvecr** [comment]
Thanks!

**Comment by Hund** [comment]
Thank you! :)

**Pull request merged**